libpod/container_internal_linux.go

```diff
@@ -174,24 +174,35 @@ func (c *Container) prepare() error {
 		return err
 	}
 
-	// Ensure container entrypoint is created (if required)
+	workdir := c.WorkingDir()
+	resolvedWorkdir, err := securejoin.SecureJoin(c.state.Mountpoint, c.WorkingDir())
+	if err != nil {
+		return errors.Wrapf(err, "error determining workdir path on container %s", c.ID())
+	}
 	if c.config.CreateWorkingDir {
-		workdir, err := securejoin.SecureJoin(c.state.Mountpoint, c.WorkingDir())
-		if err != nil {
-			return errors.Wrapf(err, "error creating path to container %s working dir", c.ID())
-		}
+		// Ensure container entrypoint is created (if required).
 		rootUID := c.RootUID()
 		rootGID := c.RootGID()
 
-		if err := os.MkdirAll(workdir, 0755); err != nil {
+		if err := os.MkdirAll(resolvedWorkdir, 0755); err != nil {
 			if os.IsExist(err) {
 				return nil
 			}
-			return errors.Wrapf(err, "error creating container %s working dir", c.ID())
+			return errors.Wrapf(err, "error creating container %s workdir", c.ID())
```

Comment:
It would probably be useful to include the value in resolveWorkdir in the error too.